src/renderer/utils/hooks.ts

```diff
@@ -1,5 +1,6 @@
 import { useDebounce } from '@react-hook/debounce';
 import { useEffect } from 'react';
+import { FunctionReturnVoid, FunctionReturnPromiseVoid } from 'sharedTypes';
```

Comment:
What's the point of these? 

`() => void` and `() => Promise<void>` are concise and used all over the place already

Reply:
You're right we mainly use `() => Promise<void>` throughout the project, however, the `nextView()` function is of type `() => void`

Reply:
That’s fine, what I mean is I don’t feel the helper types as much value

Reply:
For example you wouldn’t make a helper type

`type ArrayOfStrings = string[]`

because the name is more verbose than the actual type it’s describing. But a helper type is useful for more complex things, e.g. `MapCallback` is a good example.

I feel that FunctionReturningPromiseVoid and the other one are closer to the first category

Reply:
Notable exception to this rule: helper types that serve to clarify a specific use of a general type.

For example a selection is currently represented as a list of word indices i.e. number[] but it could be helpful to make a helper `type Selection = number[]` because otherwise number[] doesn’t really explain what it is used for

Reply:
I misread your first comment, I added these helper types after `eslint` got mad and suggested refactoring the parameter types into a helper type.

Im going to remove these types and just ignore the eslint rule.

Reply:
I've looked at it again it was a ts rule not a eslint rule ***

Reply:
Oh weird yeah, makes sense. Your solution looks fine